fix action spec for pickle

Inoryy opened the pull request:
Currently if you try to pickle output from `SC2Env::action_spec()` it will crash with a cryptic error, which turns out to be caused by pickle trying to access internal magic attribute (of the type `__name__`) in the `Functions` object and it being intercepted by `__getattr__` and subsequently crashing due to `self._func_dict` missing it.
 
So I propose a fix as per [this answer](https://stackoverflow.com/a/6416080).

A use case for using pickle in the first place is running multiple SC2Env instances and communicating with them through Pythons `multiprocessing.Process` send / recv.

tewalds commented:
I'll accept this, though with some changes for lint (consistent quotes and indentation) and for python2 compatibility (super takes args). It'd be great to have a unit test as well to make sure it keeps working.

Inoryy commented:
@tewalds fixed styling, python2 compatibility and added a unit test. 
I've also verified the issue via unit test by running it without the fix:

```
======================================================================
ERROR: test_can_pickle_action_spec (__main__.TestActionSpecPickle)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "pysc2/tests/act_spec_pickle_test.py", line 36, in test_can_pickle_action_spec
    pickle.loads(pickle.dumps(env.action_spec()))
  File "<...>/pysc2/lib/actions.py", line 334, in __getattr__
    return self._func_dict[name]
KeyError: '__getstate__'

----------------------------------------------------------------------
Ran 1 test in 9.502s

FAILED (errors=1)
```

tewalds commented:
Hmm, it'd be nice if the unit test didn't have to start the whole game. Since the action spec is actually created in features.py, I'd add it in lib/features_test.py, or even create a lib/actions_test.py. Are there any other things you're depending on being pickleable?

Inoryy commented:
@tewalds oh, didn't notice there are tests in `lib/` as well. 
Ok, I removed dependency on the game and added the test into `lib/features_test.py`. 
Didn't create `actions_test.py` because I'm also pickling obs specs, though they don't fail at the moment.

Inoryy commented:
@tewalds anything I missed?

tewalds commented:
Sorry, it's now merged internally and will be part of the next public release.
